src/identity/components/MarketoAccountUpgradeOverlay.tsx

```diff
@@ -0,0 +1,227 @@
+// Libraries
+import React, {FC, useCallback, useContext, useEffect, useState} from 'react'
+import {useDispatch, useSelector} from 'react-redux'
+import {
+  Button,
+  ComponentColor,
+  ComponentStatus,
+  Overlay,
+} from '@influxdata/clockface'
+
+// Components
+import {OverlayContext} from 'src/overlays/components/OverlayController'
+
+// Selectors
+import {selectCurrentAccountId, selectUser} from 'src/identity/selectors'
+
+// Error Reporting
+import {reportErrorThroughHoneyBadger} from 'src/shared/utils/errors'
+
+// Notifications
+import {
+  marketoFormSubmitFailure,
+  marketoFormSubmitSuccess,
+  marketoLoadFailure,
+} from 'src/shared/copy/notifications/categories/accounts-users-orgs'
+
+// Utils
+import {notify} from 'src/shared/actions/notifications'
+
+// Styles
+import './MarketoAccountUpgradeOverlay.scss'
+
+const popupLinkStyle = {
+  color: 'white',
+  textDecoration: 'underline',
+}
+
+// Types
+declare let window: WindowObjectPlusMarketoScript
+
+interface WindowObjectPlusMarketoScript extends Window {
+  MktoForms2?: MarketoScript
+}
+
+interface MarketoScript {
+  loadForm?: Function
+  getForm?: Function
+  whenReady?: Function
+}
+
+// Constants
+const BACKUP_CONTACT_SALES_LINK = 'https://www.influxdata.com/contact-sales-b/'
+const MARKETO_SERVER_INSTANCE = '//get.influxdata.com'
+const MARKETO_SUBSCRIPTION_ID = '972-GDU-533'
+const MARKETO_FORM_ID = 2826
+
+// Error Messaging
+enum MarketoError {
+  FormLoadingError = 'failed to load marketo account upgrade form',
+  FormSubmitError = 'failed to submit marketo account upgrade form',
+  ScriptFetchError = 'failed to fetch marketo account upgrade script',
+  ScriptRuntimeError = 'failed to run marketo account upgrade script',
```

Comment:
It seems like these are used as error codes, right? I was initially confused beacuse I thought these were error messages shown to the user, rather than used as error codes to determine which toast to show - I think the error names, e.g. `FormLoadingError` and `FormSubmitError` are descriptive enough for these purposes, and it might actually be slightly more confusing to treat these like error messages shown to the user, rather than just making them a code like 0, 1, 2, 3, especially since `handleError` uses the actually error that is caught.

Reply:
So, these are serving two purposes: (1) they determine which toast is sent to the user; and (2) they provide a descriptive message that goes to honeybadger (so that we can see if users are hitting a persistent issue with the form). 

Assigning the messages to an enum lets a single error handler deal with both questions - which toast am I showing the user, which is more limited - there are only two toasts - but also, which more detailed error message am I sending to honeybadger.

Reply:
I see your point though - this is a long message for something that's just being used as a honeybadger error name. let me fix this.